java/core/src/test/java/sleeper/core/schema/SchemaDuplicateFieldsTest.java

```diff
@@ -0,0 +1,96 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.core.schema;
+
+import org.junit.Test;
+import sleeper.core.schema.type.ByteArrayType;
+import sleeper.core.schema.type.IntType;
+import sleeper.core.schema.type.StringType;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class SchemaDuplicateFieldsTest {
+
+    @Test
+    public void refuseRowKeysWithSameName() {
+        // Given
+        Schema.Builder builder = Schema.builder().rowKeyFields(
+                new Field("column1", new IntType()),
+                new Field("column1", new StringType()));
+
+        // When / Then
+        assertThatThrownBy(builder::build)
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("column1");
+    }
+
+    @Test
+    public void refuseSortKeysWithSameName() {
+        // Given
+        Schema.Builder builder = Schema.builder()
+                .rowKeyFields(new Field("column1", new IntType()))
+                .sortKeyFields(
+                        new Field("column2", new StringType()),
+                        new Field("column2", new ByteArrayType()));
+
+        // When / Then
+        assertThatThrownBy(builder::build)
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("column2");
+    }
+
+    @Test
+    public void refuseValueFieldsWithSameName() {
+        // Given
+        Schema.Builder builder = Schema.builder()
+                .rowKeyFields(new Field("column1", new IntType()))
+                .valueFields(
+                        new Field("column2", new StringType()),
+                        new Field("column2", new ByteArrayType()));
+
+        // When / Then
+        assertThatThrownBy(builder::build)
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("column2");
+    }
+
+    @Test
+    public void refuseFieldNameInRowKeysAndSortKeys() {
+        // Given
+        Schema.Builder builder = Schema.builder()
+                .rowKeyFields(new Field("column1", new IntType()))
+                .sortKeyFields(new Field("column1", new StringType()));
+
+        // When / Then
+        assertThatThrownBy(builder::build)
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("column1");
+    }
+
+    @Test
+    public void refuseFieldNameInSortKeysAndValueKeys() {
+        // Given
+        Schema.Builder builder = Schema.builder()
+                .rowKeyFields(new Field("column1", new IntType()))
+                .sortKeyFields(new Field("column2", new StringType()))
+                .valueFields(new Field("column2", new ByteArrayType()));
+
+        // When / Then
+        assertThatThrownBy(builder::build)
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("column2");
+    }
+}
```

Comment:
For completeness, can we add a test refuseFieldNameInRowKeysAndValueKeys ?

Reply:
Done